src/language/markdown-source-code.js

```diff
@@ -70,6 +190,79 @@ export class MarkdownSourceCode extends TextSourceCodeBase {
 		return this.#parents.get(node);
 	}
 
+	/**
+	 * Returns an array of all inline configuration nodes found in the
+	 * source code.
+	 * @returns {Array<InlineConfigComment>} An array of all inline configuration nodes.
+	 */
+	getInlineConfigNodes() {
+		if (this.#inlineConfigComments.length === 0) {
+			this.#inlineConfigComments = this.#htmlNodes.flatMap(
+				extractInlineConfigCommentsFromHTML,
+			);
+		}
```

Comment:
This would search for inline config comments every time if there are none (i.e., `this.#inlineConfigComments` ends up remaining an empty array after this). I think it's better to not initialize `this.#inlineConfigComments` and just check `if (!this.#inlineConfigComments)` here. That would also catch errors if some other code tries to access `this.#inlineConfigComments` directly.

Reply:
Good call.